src/Common/src/TypeSystem/CodeGen/TargetDetails.CodeGen.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.TypeSystem
+{
+    // Extension to TargetDetails related to code generation
+    partial class TargetDetails
+    {
+        public TargetDetails(TargetArchitecture architecture, TargetOS targetOS, TargetAbi abi, MaximumSimdVectorLength simdVectorLength)
+            : this(architecture, targetOS, abi)
+        {
+            MaximumSimdVectorLength = simdVectorLength;
+        }
+
+        public MaximumSimdVectorLength MaximumSimdVectorLength
+        {
+            get;
```

Comment:
I don't really like that idea. Its very much readable to have a switch case arrangement around use of target architecture, and while it doesn't cover all cases, its pretty good. Forcing the use of a series of helper functions is a bit of a kludge.